api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -184,13 +184,13 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 			}
 			
 			//concept should be the same as on previous order, same applies to drug for drug orders
-			boolean isDrugOrderAndHasADrug = isDrugOrder && ((DrugOrder) order).getDrug() != null;
+			boolean isDrugOrderAndHasADrug = isDrugOrder && (((DrugOrder) order).getDrug() != null || ((DrugOrder) order).isNonCodedDrug()) ;
```

Comment:
This change is not required, an order with a non coded drug value but that has a blank drug field  should be invalid